src/linear_eq.jl

```diff
@@ -103,10 +103,13 @@ function gauss_elimination_interval!(x::AbstractArray, A::AbstractMatrix, b::Abs
 
     if precondition
         (A, b) = preconditioner(A, b)
-    else
-        A = copy(A)
-        b = copy(b)
     end
+    _A = A
+    _b = b
+    A = similar(A)
+    b = similar(b)
+    A .= _A
+    b .= _b
```

Comment:
Is this different / better than `_A = A; A = copy(_A)`?

Reply:
If I recall correctly, calling `similar()` on a `StaticArray` gives a `MutableArray` which is required for the `setindex!()` calls below.